kratos/geometries/geometry.h

```diff
@@ -527,6 +527,16 @@ class Geometry
         return this->size();
     }
 
+    /// Returns number of points per direction.
+    virtual SizeType PointsNumberInDirection(IndexType DirectionIndex) const
+    {
+        KRATOS_DEBUG_ERROR_IF(DirectionIndex > 2) << "Possible direction index reaches from 0-2. "
```

Comment:
It would be better to send an error here in the base class and only implement it in the ones that make sense.